server/src/main/java/com/lunchtime/seed/DatabaseSeed.java

```diff
@@ -0,0 +1,83 @@
+package com.lunchtime.seed;
+
+import com.lunchtime.models.Feedback;
+import com.lunchtime.models.Restaurant;
+import com.lunchtime.repository.FeedbackRepository;
+import com.lunchtime.repository.RestaurantRepository;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import java.sql.Date;
+import java.time.Instant;
+import java.util.List;
+
+
+@Component
+public class DatabaseSeed {
+
+    String[] restaurantName = new String[] {"Avocado", "Masuri", "Gavana", "Cake", "Green", "Nuts"};
+    String[] userName = new String[] {"Bob", "Devid", "Tom", "Alan", "Leo", "Fred"};
+    Float[] cordLatitude = new Float[] { 50.616294f, 50.618261f, 50.620219f, 50.616146f, 50.618318f, 50.624449f };
+    Float[] cordLongitude = new Float[] { 26.275728f, 26.260064f, 26.241863f, 26.253994f, 26.252249f, 26.249677f };
+
+    private final RestaurantRepository restaurantRepository;
+    private final FeedbackRepository feedbackRepository;
+
+    public DatabaseSeed(RestaurantRepository restaurantRepository, FeedbackRepository feedbackRepository) {
+        this.restaurantRepository = restaurantRepository;
+        this.feedbackRepository = feedbackRepository;
+    }
+
+    @EventListener
+    public void seed(ContextRefreshedEvent event) {
+        if (restaurantRepository.count() == 0L) {
+            seedRestaurant();
+        }
+
+        if (feedbackRepository.count() == 0L) {
+            seedFeedback();
+        }
+
+    }
+
+    public void seedRestaurant() {
+
+        for (Long i = Long.valueOf(1); i < restaurantName.length; i++) {
+
+            Restaurant rest = new Restaurant(restaurantName[i.intValue()],
+                restaurantName[i.intValue()].concat("@gmail.com"),
+                "Rivne, street ".concat(i.toString()),
+                "www.".concat(restaurantName[i.intValue()]).concat(".ua"),
+                "Description for restaurant ".concat(restaurantName[i.intValue()]),
+                "12-24", false,
+                i, i, i.intValue(), cordLongitude[i.intValue()],
+                cordLatitude[i.intValue()], Instant.now(),
+                i, Instant.now(), i);
+
+            restaurantRepository.save(rest);
+        }
+
+    }
+
+    public void seedFeedback() {
+
+        List<Restaurant> res = restaurantRepository.findAll();
+
+        for (Long i = Long.valueOf(1); i < restaurantName.length; i++) {
+
+            Feedback feedback = new Feedback(
+                "User ".concat(userName[i.intValue()]).concat(" write comment to restaurant"),
+                true,
+                new Date(System.currentTimeMillis()),
+                userName[i.intValue()],
+                res.get(i.intValue() - 1),
+                3 + i.intValue(),
+                12 + i.intValue());
+
+            feedbackRepository.save(feedback);
+        }
```

Comment:
Unfortunately, I had no possibilities to understand this code before. According to my simple survey I researched that JPA has got its **[own way](https://docs.spring.io/spring-boot/docs/current/reference/html/howto.html#howto-database-initialization)** for database initialization. 

Since we are going to use this till our project is finished, I suppose we need more convenient way for our purpose to fill database. I find **[this](https://github.com/kishanjavatrainer/SpringBootJPADatabaseInitialization)** way more relevant.

Reply:
What's wrong in code? I use JPA to save/find method and we Hibernate

Reply:
Don't worry, nothing is wrong. Just mentioning about JPA easier way to **[initialize](https://github.com/kishanjavatrainer/SpringBootJPADatabaseInitialization/blob/master/SpringBootJPADatabaseInitialization/src/main/java/com/infotech/app/SpringBootJPADatabaseInitialization.java)** database.